### DISCUSSION TITLE
Global Dependency Dashboard

### DISCUSSION THREAD

**Discussion by Mitgenosse** [renovatebot/renovate#33219] [Suggest an Idea]
### Tell us more.

As part of a security hackathon at our company last year I wanted to create a "global" dependency dashboard. This dashboard included dependency information of all projects managed by our Renovate GitLab runner. The idea was to have an overview about all dependency versions across the repos to identify which repos use certain outdated or vulnerable dependencies.

## Specification

The idea was to include 

* an overview about all repositories managed by the Renovate runner
* links to the detailed "Dependency Dashboards" of each project
* a filter for package managers ("show me all java/maven projects")
* a search field to find all projects using a dependency

## How it has been done (hacked)

Since renovate didn't provide any interface providing such information (maybe it does now) I simply relied on fetching the dependency dashboards of every project by fetching the list of issues created by my Renovate user. This happened after every pipeline run of Renovate. It fetches all issue descriptions, extracts the dependencies from the text, formats it and dumps it into a local json file. Then we used GitLab pages to upload that json file, along with some html and js script to render it. All projects are listed, you can expand the dependencies section to see all the dependencies of a project's manager and you have a direct link to the Dependency Dashboard. We didn't get to the filtering and searching part, but I'm happy to say we won the hackathon anyways.

<img width="1458" alt="Screenshot 2024-12-20 at 13 01 34" src="https://github.com/user-attachments/assets/c5555902-3900-4898-89dc-3941152d6b62" />

## Potential proper solution

IF (big if) this is something the community is interested in, a proper solution could be to define a standard report model and have Renovate export that as part of the Renovate execution itself. Either by a single report standard, or a plugin system supporting custom reporters.

If it's a single report standard as a .json file, external web-apps could consume it to implement the specification. Otherwise a custom reporter could also directly build an html report which can be uploaded to something like GitLab pages.

I would be interested in supporting the development. Looking forward to your opinion.

**Comment by secustor** [comment]
Take a look at this ( unofficial ) [Renovate Backstage plugin](https://github.com/secustor/backstage-plugins/tree/main/plugins/renovate), if you need something like that.
